app/main.py

```diff
@@ -1,25 +1,66 @@
 from abc import ABC
+from typing import Type, Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.name = name
+
+    def __set__(self, instance: Any, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value must be an integer.")
+
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"Value must be between "
+                             f"{self.min_amount} and "
+                             f"{self.max_amount}.")
+        instance.__dict__[self.name] = value
+
+    def __get__(self, instance: int, owner: Any) -> None:
```

Comment:
The return type hint for the `__get__` method should be `Any` instead of `None`. This method is expected to return the value associated with the descriptor, not `None`.